### DISCUSSION TITLE
Automatic batch processing for certain settings

### DISCUSSION THREAD

**Discussion by KannManMachen** [lllyasviel/Fooocus#2510] [Q&A] [answered]
Hey there, is it possible to configure and run certain setups (with fixed and variable values) as batch processing?

Specifically: I would like to automatically process the (whole number) values of the "Guidance Scale" from 0 - 30 in my setup without having to manually adjust the value after each generation.

Greetings

**Comment by mashb1t** [comment, marked as answer]
This currently is only possible using the API. You can find the definition in the footer.

**Reply by KannManMachen** [reply]
Hey @mashb1t,
thanks for the tip :)

I have set up the API from @[mrhan1993](https://github.com/mrhan1993/Fooocus-API).

I have looked at the different endpoints (via the fastAPI UI) and tried a few. Everything works. However, I still don't understand how to get the desired batch processing to work with the help of this API.

But I had also described my issue again in parallel with @mrhan1993 in his [Github](https://github.com/mrhan1993/Fooocus-API/issues/243). As soon as I'm smarter, I'll report back here.

Cheers.


P.S.: about my background – I'm not a developer and working with APIs is still pretty new territory for me. But I am curious and willing to learn ;)

**Reply by KannManMachen** [reply]
Hey @mashb1t,

apparently there seems to be no "easy" way to trigger the desired batch processing with the help of / on the basis of the API - at least not without programming knowledge.

I have now come across the "function / tool" that I have been looking for all this time. It's the "Prompt Matrix / XYZ Plot grid", which you can apparently find in Automatic1111:

[https://github.com/AUTOMATIC1111/stable-diffusion-webui/blob/master/scripts/xyz_grid.py](https://github.com/AUTOMATIC1111/stable-diffusion-webui/blob/master/scripts/xyz_grid.py)

I can use this to link various parameters (steps, CFG scale, samplers, styles, checkpoint name, etc.)  
) with certain value methods (Comma Separated, Simple Range, Divided Range, Increment, Decrement).This will save me a lot of time when searching for certain setups.

So the journey continues (DALL-E 3 > Midjourney > Draw Things > Fooocus > Automatic1111 ...). Thank goodness I will still be able to control the majority with Fooocus <3

Cheers
